recipe-server/client/control/components/RecipeContainer.js

```diff
@@ -17,46 +22,88 @@ export default function composeRecipeContainer(Component) {
       }
     }
 
-    componentWillReceiveProps({ recipeId }) {
-      if (recipeId && recipeId !== this.props.recipeId) {
-        this.getRecipeData(recipeId);
+    componentWillReceiveProps({ recipeId, routeParams = {} }) {
+      const isRecipeChanging = recipeId !== this.props.recipeId;
+      const isRouteRevisionChanging =
+        routeParams.revisionId !== this.props.routeParams.revisionId;
+
+      if (recipeId && isRecipeChanging) {
+        this.getRecipeData(recipeId, routeParams && routeParams.revisionId);
+      }
+      if (routeParams.revisionId && isRouteRevisionChanging) {
+        this.getRecipeRevision(routeParams.revisionId);
       }
     }
 
-    getRecipeData(recipeId) {
-      const { dispatch, location, recipe } = this.props;
+    getRecipeData(recipeId, revisionId) {
+      const { dispatch, routeParams, recipe } = this.props;
       if (!recipe) {
         dispatch(setSelectedRecipe(recipeId));
 
-        if (location.query.revisionId) {
-          dispatch(makeApiRequest('fetchSingleRevision', { revisionId: location.query.revisionId }))
-          .then(revision => {
-            dispatch(singleRecipeReceived(revision.recipe));
-          });
-        } else {
-          dispatch(makeApiRequest('fetchSingleRecipe', { recipeId }))
-          .then(newRecipe => {
-            dispatch(singleRecipeReceived(newRecipe));
-          });
-        }
+        // always get the latest
+        dispatch(makeApiRequest('fetchSingleRecipe', { recipeId }))
+        .then(newRecipe => {
+          dispatch(singleRecipeReceived(newRecipe));
+          // get a specific revision, if specified
+          if (revisionId || routeParams.revisionId) {
+            this.getRecipeRevision(revisionId || routeParams.revisionId);
+          }
+        });
       }
     }
 
+    getRecipeRevision(revisionId) {
+      const { dispatch } = this.props;
+
+      return dispatch(makeApiRequest('fetchSingleRevision', { revisionId }))
+        .then(revision => {
+          dispatch(singleRevisionReceived({
+            revision,
+            recipeId: revision.recipe.id,
```

Comment:
Do you mean attaching the `recipeId` prop? Or the actual `singleRevisionReceived` call?